### ISSUE TITLE
The last message on the scaler is blocked

### ISSUE THREAD

**Issue by amitherlekar** [kedacore/keda#1936]
### Report

Basically, I want spin up a pod for every message. But the maximum number pods at a time is 2. 

### Expected Behavior

Case A: If the number of messages = 6, then pods creation order should be [1 ,3, 5] and [ 2 , 4 , 6]
Case B: If the number of messages = 3, then pods creation order should be [1, 3]  and  [2] 

### Actual Behavior

Case A: Pods [1,  3,  5] and  [2, 4] spin up correctly without blocking whereas,  6th pod  starts only after the 5th pod completes execution.

Case B: Pods 1 and 2  spin up correctly without blocking whereas,  3rd pod will starts only after the 2nd pod completes execution.

Basically the (n-1)th message is always blocked (waiting to be consumed), where n is number of messages.

### Steps to Reproduce the Problem

The ScaledJob configuration

pollingInterval: 1
parallelism: 1 
completions: 1
maxReplicaCount: 2 
queueLength: 1

### Logs from KEDA operator

```
example
```

### KEDA Version

2.3.0

### Kubernetes Version

1.20

### Platform

Any

### Scaler Details

ActiveMQ Artemis

**Comment by zroubalik** [comment]
@TsuyoshiUshio FYI^
